src/main/java/reposense/parser/CsvParser.java

```diff
@@ -51,105 +51,106 @@ public CsvParser(Path csvFilePath) throws IOException {
-                String[] elements = line.split(ELEMENT_SEPARATOR);
+        try {
+            csvReader = new FileReader(this.csvFilePath.toFile());
+            records = CSVFormat.DEFAULT.withFirstRecordAsHeader().parse(csvReader);
+        } catch (IOException ioe) {
+            throw new IOException(MESSAGE_UNABLE_TO_READ_CSV_FILE, ioe);
+        }
 
-                if (line.isEmpty() || isLineMalformed(elements, lineNumber, line)) {
-                    continue;
-                }
+        for (CSVRecord record : records) {
+            if (isLineMalformed(record)) {
+                continue;
+            }
 
-                try {
-                    processLine(results, elements);
-                } catch (ParseException pe) {
-                    logger.warning(String.format(MESSAGE_LINE_PARSE_EXCEPTION_FORMAT,
-                            lineNumber, csvFilePath.getFileName(), line, pe.getMessage()));
-                }
+            try {
+                processLine(results, record);
+            } catch (ParseException pe) {
+                logger.warning(String.format(MESSAGE_LINE_PARSE_EXCEPTION_FORMAT,
+                        getLineNumber(record), csvFilePath.getFileName(), pe.getMessage()));
+            } catch (IllegalArgumentException iae) {
+                logger.log(Level.WARNING, iae.getMessage(), iae);
             }
-        } catch (IOException ioe) {
-            throw new IOException(MESSAGE_UNABLE_TO_READ_CSV_FILE, ioe);
-        } catch (IllegalArgumentException iae) {
-            logger.log(Level.WARNING, iae.getMessage(), iae);
         }
 
         return results;
     }
 
     /**
-     * Checks if the {@code line} contains values at the mandatory positions in CSV format.
+     * Returns true if {@code record} does not contain values at the mandatory columns in CSV format.
      */
-    private boolean isLineMalformed(final String[] elements, int lineNumber, String line) {
+    private boolean isLineMalformed(CSVRecord record) {
         for (int position : mandatoryPositions()) {
-            if (!containsValueAtPosition(elements, position)) {
-                logger.warning(String.format(MESSAGE_MALFORMED_LINE_FORMAT,
-                        lineNumber, csvFilePath.getFileName(), line));
+            if (record.get(position).isEmpty()) {
+                logger.warning(String.format(MESSAGE_MALFORMED_LINE_FORMAT, getLineNumber(record),
+                        csvFilePath.getFileName()));
                 return true;
             }
         }
-
         return false;
     }
 
     /**
-     * Checks that {@code position} in within the range of {@code element} array and
-     * value in {@code position} is not empty.
+     * Returns the value of {@code record} at {@code colNum}.
      */
-    private boolean containsValueAtPosition(final String[] elements, int position) {
-        return elements.length > position && !elements[position].isEmpty();
```

Comment:
Is it safe to access a non-existing column? i.e. `record.get(-1)`

Maybe we can use [`CsvRecord#isConsistent`](https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVRecord.html#isConsistent--) to replicate `elements.length > position` behavior?

Reply:
I considered using [`CsvRecord#isConsistent`](https://commons.apache.org/proper/commons-csv/apidocs/org/apache/commons/csv/CSVRecord.html#isConsistent--) but decided not to due to the test case [CsvParserTest#merge_emptyLocation_success](https://github.com/reposense/RepoSense/blob/4c915c422e30b20f8c1674c036dc2c7a984af162/src/test/java/reposense/parser/CsvParserTest.java#L256) .

Apparently the test file `repo-config.csv` is corrupted, where it's supposed to be 6 columns normally but this only has 5 columns (missing ignore commit hash column).
https://github.com/reposense/RepoSense/blob/4c915c422e30b20f8c1674c036dc2c7a984af162/src/test/resources/CsvParserTest/repoconfig_merge_empty_location_test/repo-config.csv .

Should I fix the CSV file in question?

Reply:
For such case, just update the csv. We may have missed it 😨.

Reply:
Ok. Will use `isConsistent` method and fix the CSV file.

Reply:
> Is it safe to access a non-existing column? i.e. `record.get(-1)`

In any case, is there a reason why we need to take `get(-1)`  into account?

Reply:
Its just an illustration, for cases where columns are missing. 
i.e. `record.get(5)` but there's only `4` columns.

`CsvRecord#isConsistent` should be an elegant approach to handle this.

Just make sure not to change the behavior of the parser, we skip the erroneous lines without terminating the program.

Reply:
> Just make sure not to change the behavior of the parser, we skip the erroneous lines without terminating the program.

Yes that's the current behavior